pandas/tests/tseries/test_timezones.py

```diff
@@ -1280,6 +1280,25 @@ def test_ambiguous_compat(self):
             self.assertEqual(result_pytz.to_pydatetime().tzname(),
                              result_dateutil.to_pydatetime().tzname())
 
+    def test_tzreplace_issue_15683(self):
+        """Regression test for issue 15683."""
+        dt = datetime(2016, 3, 27, 1)
+        tzinfo = pytz.timezone('CET').localize(dt, is_dst=False).tzinfo
+
+        result_dt = dt.replace(tzinfo=tzinfo)
+        result_pd = Timestamp(dt).replace(tzinfo=tzinfo)
+
+        self.assertEqual(result_dt.timestamp(), result_pd.timestamp())
```

Comment:
We've recently switched to pytest for our test-runner, so you can write these as

```python
assert result_dt.timesamp() == result_pd.timestamp()
...
```